lib/rspec/support/method_signature_verifier.rb

```diff
@@ -158,6 +158,15 @@ def unlimited_args?
       end
 
       INFINITY = 1 / 0.0
+
+      private
+
+      def method_or_class_initialize(method)
+        # Support only not redefined Class.new
+        return method if !method.respond_to?(:owner) || method.owner != Class
+        return method if !method.respond_to?(:name) || method.name != :new
```

Comment:
What are the cases when methods don't respond to `owner` or `name`?

Reply:
At least one of the tests are passing Proc, which fails without this check.